spec/enumerable_spec.rb

```diff
@@ -0,0 +1,213 @@
+
+    it 'if pattern supplied, returns true if pattern equals at least one element in a collection' do
+      expect(arr_string.my_any?(/t/)).to be true
+    end
+
+    it 'if regex given, returns false if it is not equal to any element in a collection' do
+      expect(arr_string.my_any?(/[0-9]/)).to be false
+    end
+
+    it 'if class is passed as an argument returns true if at least one of the collection is a member of such class' do
+      expect(arr_nil.my_any?(Integer)).to be true
+    end
+
+    it 'if class is passed as an argument returns false if none of the collection is a member of such class' do
+      expect(arr_string.my_any?(Integer)).to be false
+    end
+  end
+
+  describe '#my_none?' do
+    it 'passes elements of the collection to the block and returns true if the block never returns true' do
+      expect(arr_string.my_none? { |x| x.instance_of?(Integer) }).to be true
+    end
+
+    it 'passes elements of the collection to the block and returns false if the block returns true' do
+      expect(arr.my_none? { |x| x == 1 }).to be false
+    end
+
+    it 'if block not given, returns true if none of the collection members is true' do
+      expect(arr_none.my_none?).to be true
+    end
+
+    it 'if block not given, returns false if at least one of the collection members is true' do
+      expect(arr_false.my_none?).to be false
+    end
+
+    it "if pattern supplied, the returns true if none of collection's members matches pattern" do
+      expect(arr_string.my_none?(/t/)).to be false
+    end
+
+    it "if regex supplied, the returns true if none of collection's members matches it" do
+      expect(arr_string.my_none?(/[0-9]/)).to be true
+    end
+
+    it 'if class is passed as an argument returns true if none of the collection is a member of such class' do
+      expect(arr_none.my_none?(Integer)).to be true
+    end
+
+    it 'if class passed, returns false if no element in collection is a member of such class' do
+      expect(arr_nil.my_none?(Integer)).to be false
+    end
+  end
+
+  describe '#my_count' do
+    it 'returns a number of elements in collection, if no argument or block given' do
+      expect(arr.my_count).to eql(5)
+    end
+
+    it 'if argument is given, returns the number of items in collection that are equal argument' do
+      expect(arr.my_count(4)).to eql(1)
+    end
+
+    it 'if block is given, counts the number of elements yielding a true value' do
+      expect(arr.my_count { |x| x % 2.positive? }).to be(3)
+    end
+  end
+
+  describe '#my_map' do
+    it 'returns a new array with the results of running block for every element in collection, if block given' do
+      expect(arr.my_map { |x| x * 2 }).to eql([2, 4, 6, 8, 10])
+    end
+
+    it 'if proc given, returns a new array with the results of running proc for every element in collection' do
+      expect(arr.my_map(proc { |x| x * 2 })).to eql([2, 4, 6, 8, 10])
+      expect(arr.my_map(&:to_s)).to eql(%w[1 2 3 4 5])
+    end
```

Comment:
* [ ] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

Reply:
Done!